src/main/scala/com/redislabs/provider/redis/redisFunctions.scala

```diff
@@ -441,6 +466,23 @@ object RedisContext extends Serializable {
     pipeline.sync()
     conn.close()
   }
+
+  /**
+    * @param arr HLL kvps to be updated in target host
+    *            calls PFADD key value to each pair in RDD
+    */
+  def setPF(arr: Iterator[(String, String)], redisConfig: RedisConfig): Unit = {
+    arr.map(kv => (redisConfig.getHost(kv._1), kv)).toArray.groupBy(_._1).
+      mapValues(a => a.map(p => p._2)).foreach {
+      x => {
+        val conn = x._1.endpoint.connect()
```

Comment:
Please refactor to use `foreachWithPipeline` utility function, see `setKVs` for example.